### PR TITLE
Vertx with Dagger showcase

### PR DESCRIPTION
**Pull request by jzillmann**
* Use modules to separate route definitions and server setup

### CODE REVIEW

**Review by arouel — APPROVED** [review]

File: src/main/java/prototype/vertx_plus_dagger/VertxServerModule.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import dagger.Module;
+import dagger.Provides;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+
+@Module
+public class VertxServerModule {
```
Comment (arouel) [inline src/main/java/prototype/vertx_plus_dagger/VertxServerModule.java]: I would prefer calling this `VertxServer`. Can a module marked `final`?

File: src/main/java/prototype/vertx_plus_dagger/http/CommonHttpEndpoints.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import prototype.vertx_plus_dagger.RouteSetup;
+
+@Module
+public class CommonHttpEndpoints {
+
+    @Provides
+    @Singleton
+    @IntoSet
+    RouteSetup root() {
+        return (router) -> router.route(HttpMethod.GET, "/")
```
Comment (arouel) [inline src/main/java/prototype/vertx_plus_dagger/http/CommonHttpEndpoints.java]: `(router)` can also be `router`

File: src/main/java/prototype/vertx_plus_dagger/http/UserHttpEndpoints.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import prototype.vertx_plus_dagger.RouteSetup;
+
+@Module
+public class UserHttpEndpoints {
+
+    @Provides
+    @Singleton
+    @IntoSet
+    RouteSetup user() {
+        return (router) -> router.route(HttpMethod.GET, "/user/:username")
```
Comment (arouel) [inline src/main/java/prototype/vertx_plus_dagger/http/UserHttpEndpoints.java]: `(router)` can also be `router`

**Review by arouel — CHANGES_REQUESTED** [review]

File: src/main/java/prototype/vertx_plus_dagger/RouteSetup.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package prototype.vertx_plus_dagger;
+
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+
+@FunctionalInterface
+public interface RouteSetup {
```
Comment (arouel) [inline src/main/java/prototype/vertx_plus_dagger/RouteSetup.java]: Would you like the name `RouteDefinition` as well?

Comment (jzillmann) [reply src/main/java/prototype/vertx_plus_dagger/RouteSetup.java]: ;) That was the name before.

**Review by arouel — APPROVED** [review]

**Pull request merged**